### ISSUE TITLE
(Unsupported SQL column type) Error - EvoMaster process terminated abruptly

### ISSUE THREAD

**Issue by mahayyuakbarisena** [EMResearch/EvoMaster#253]
Hi, 
I'm trying evomaster on my project, I run evomaster.jar which is downloaded from 
[link](https://github.com/EMResearch/EvoMaster/releases/download/v1.1.0/evomaster.jar) with the command 

> java -jar evomaster.jar - maxTime 150s --outputFolder [directory-name] --outputFormat JAVA_JUNIT_5 

and get the following error

```
* [ERROR] EvoMaster process terminated abruptly. This is likely a bug in EvoMaster. Please copy&paste the following stacktrace, and create a new issue on https://github.com/EMResearch/EvoMaster/issues
java.lang.reflect.InvocationTargetException: invokedynamic: method=public void org.evomaster.core.problem.rest.service.AbstractRestSampler.initialize(), target=org.evomaster.core.problem.rest.service.RestSampler@2f6e28bc
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:316)
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:298)
        at com.netflix.governator.lifecycle.LifecycleManager.startInstance(LifecycleManager.java:266)
        at com.netflix.governator.lifecycle.LifecycleManager.add(LifecycleManager.java:175)
        at com.netflix.governator.guice.InternalLifecycleModule.onProvision(InternalLifecycleModule.java:81)
        at com.google.inject.internal.ProvisionListenerStackCallback$Provision.provision(ProvisionListenerStackCallback.java:120)
        at com.google.inject.internal.ProvisionListenerStackCallback.provision(ProvisionListenerStackCallback.java:66)
        at com.google.inject.internal.ConstructorInjector.construct(ConstructorInjector.java:93)
        at com.google.inject.internal.ConstructorBindingImpl$Factory.get(ConstructorBindingImpl.java:306)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.FactoryProxy.get(FactoryProxy.java:62)
        at com.google.inject.internal.ProviderToInternalFactoryAdapter.get(ProviderToInternalFactoryAdapter.java:40)
        at com.google.inject.internal.SingletonScope$1.get(SingletonScope.java:168)
        at com.google.inject.internal.InternalFactoryToProviderAdapter.get(InternalFactoryToProviderAdapter.java:39)
        at com.google.inject.internal.InternalInjectorCreator.loadEagerSingletons(InternalInjectorCreator.java:213)
        at com.google.inject.internal.InternalInjectorCreator.injectDynamically(InternalInjectorCreator.java:184)
        at com.google.inject.internal.InternalInjectorCreator.build(InternalInjectorCreator.java:111)
        at com.google.inject.internal.InjectorImpl.createChildInjector(InjectorImpl.java:246)
        at com.netflix.governator.guice.LifecycleInjector.createChildInjector(LifecycleInjector.java:331)
        at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:411)
        at com.netflix.governator.guice.LifecycleInjector.createInjector(LifecycleInjector.java:352)
        at org.evomaster.core.Main$Companion.init(Main.kt:236)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:139)
        at org.evomaster.core.Main$Companion.main(Main.kt:72)
        at org.evomaster.core.Main.main(Main.kt)
Caused by: java.lang.IllegalArgumentException: Column data type TIME is not supported in Evomaster Data types
        at org.evomaster.core.database.schema.ColumnFactory.parseColumnDataType(ColumnFactory.kt:34)
        at org.evomaster.core.database.schema.ColumnFactory.createColumnFromDto(ColumnFactory.kt:14)
        at org.evomaster.core.database.SqlInsertBuilder.<init>(SqlInsertBuilder.kt:99)
        at org.evomaster.core.problem.rest.service.RestSampler.initSqlInfo(RestSampler.kt:30)
        at org.evomaster.core.problem.rest.service.AbstractRestSampler.initialize(AbstractRestSampler.kt:84)
        at com.netflix.governator.lifecycle.LifecycleMethods.methodInvoke(LifecycleMethods.java:313)
        ... 26 common frames omitted
```

**Comment by arcuri82** [comment]
Hi,
 thx for trying out EM.  

Which database are you using? is it MySQL?

**Comment by arcuri82** [comment]
anyway, meanwhile u can try to use ` --extractSqlExecutionInfo false --generateSqlDataWithSearch false --heuristicsForSQL false` to disable SQL handling

**Comment by mahayyuakbarisena** [comment]
Hi, thx for your response, I'm trying to disable SQL handling and get another error : 
```
* [ERROR] EvoMaster process terminated abruptly. This is likely a bug in EvoMaster. Please copy&paste the following stacktrace, and create a new issue on https://github.com/EMResearch/EvoMaster/issues
java.util.NoSuchElementException: Key  is missing in the map.
        at kotlin.collections.MapsKt__MapWithDefaultKt.getOrImplicitDefaultNullable(MapWithDefault.kt:24)
        at kotlin.collections.MapsKt__MapsKt.getValue(Maps.kt:341)
        at org.evomaster.core.output.ObjectGenerator.getNamedReference(ObjectGenerator.kt:99)
        at org.evomaster.core.output.oracles.SchemaOracle.supportedObject(SchemaOracle.kt:133)
        at org.evomaster.core.output.oracles.SchemaOracle.generatesExpectation(SchemaOracle.kt:234)
        at org.evomaster.core.output.PartialOracles.activeOracles(PartialOracles.kt:164)
        at org.evomaster.core.problem.rest.service.AbstractRestFitness.handleResponseTargets(AbstractRestFitness.kt:366)
        at org.evomaster.core.problem.rest.service.RestFitness.doCalculateCoverage(RestFitness.kt:112)
        at org.evomaster.core.problem.rest.service.RestFitness.doCalculateCoverage(RestFitness.kt:22)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage(FitnessFunction.kt:59)
        at org.evomaster.core.search.service.FitnessFunction.calculateCoverage$default(FitnessFunction.kt:46)
        at org.evomaster.core.search.algorithms.MioAlgorithm.search(MioAlgorithm.kt:38)
        at org.evomaster.core.Main$Companion.run(Main.kt:284)
        at org.evomaster.core.Main$Companion.initAndRun(Main.kt:150)
        at org.evomaster.core.Main$Companion.main(Main.kt:72)
        at org.evomaster.core.Main.main(Main.kt)
```
I'm using hibernate database (h2) for configuring EMDriver, following your tutorial from this [link](https://youtu.be/ORxZoYw7LnM)

**Comment by arcuri82** [comment]
thx

@bogdanmarculescu as this seems a bug in the PartialOracles, can you look at it please?

**Comment by arcuri82** [comment]
this should now be fixed in the new 1.2.0 release. if not, please open a new issue
